accepted/0000-salt-new-maintaining-workflow.md

```diff
@@ -0,0 +1,300 @@
+
+### Packaging artifacts
+
+All current extra "Sources" files in our RPM package will go now to a `pkg/suse/` directory in `openSUSE/salt`, together with the spec file, the different maintained changelogs files:
+
+```
+pkg/suse/README.SUSE
+pkg/suse/html.tar.bz2
+pkg/suse/salt-tmpfiles.d
+pkg/suse/transactional_update.conf
+pkg/suse/update-documentation.sh
+pkg/suse/mkchlog.sh
+pkg/suse/_multibuild
+pkg/suse/salt.spec
+pkg/suse/changelogs/factory/salt.changes
+pkg/suse/changelogs/sles15sp2/salt.changes
+pkg/suse/changelogs/sles15sp3/salt.changes
+pkg/suse/changelogs/sles15sp4/salt.changes
+pkg/suse/changelogs/sles15sp5/salt.changes
+```
+
+This is the place now where all those files will be maintained.
+
+#### Tracking upstream and downstream patches
+
+This RFC proposes the usage of "obs_scm" service as the mechanism to pull the sources for the Salt package to build. This service will automatically produce a sources tarball according to a configured GitHub branch.
+
+For the regular Salt maintenance, this means it won't be needed anymore to manually produce patch files to add them to the spec file, as the tarball now contains the updated sources (with the exception of EMBARGOED bugs, where patches are still needed as we cannot push any fix to public GitHub repositories).
+
+To avoid losing the useful labeling of "PATCH-FIX_UPSTREAM" and "PATCH-FIX_OPENSUSE" (with a direct link to the origin PR on the spec file for each new patch we introduced into our Salt package), we will keep adding this information to the spec file on every new PR but this time without adding the patch itself, only the comment.
+
+#### Salt RPM changelogs
+
+As mentioned, the changelog files are now maintained in the `openSUSE/salt` GitHub repo, under `pkg/suse/changelogs/` directory.
+
+Our packaging artifacts will contain a `mkchlog.sh`, which is a helper script to generate a changelog entry to all maintained changelog in one shot. Something like this:
+
+```bash
+echo "Generating changelog entry for Salt package"
+if ! osc vc _temp.changes;
+then
+    exit 1;
+fi
+
+echo "Update changelog files"
+echo >> _temp.changes
+
+for i in $(ls changelogs/*/salt.changes); do
+    echo "$(cat _temp.changes $i)" > $i
+    git add $i
+done
+
+rm _temp.changes
+```
+
+When creating a PR to `openSUSE/salt` the user must also include the corresponding changelog entry for all maintained changelog files.
+
+Similarly to the main Uyuni repository, we should add a GitHub action to warn the user in case no changelog entry is added in the PR.
+
+NOTE: I think it is better to decouple commit messages (focus on developers) from changelog entries (focus on users/customers), so I prefer to not use commit messages from "openSUSE/salt" to autogenerate the changelog entries but rather to manually write a meaningful changelog message to be included in your PR as part of your changes. Similarly to what we do in other Uyuni repositories.
```

Comment:
Totally agree. In openSUSE/salt, we can't afford to lose meaningful commit messages